### PR TITLE
Add precommit and linters

### PR DESCRIPTION
**Pull request by pat-s**
same configs as in the main repo

Maybe the plugin should also switch from `*` to `-` for md lists?

### CODE REVIEW

File: src/utils/__fixtures__/CHANGELOG_2.md [hunk 1]
```diff
@@ -2,84 +2,84 @@
...
-  * Regognize "CI_*" EnvVars (#6)
-* ENHANCEMENTS
-  * Multiarch build (#8)
-* MISC
-  * Upgrade urfave/cli to v2 (#5)
\ No newline at end of file
+- FEATURES
+  - Add ppc64le support (#8)
+- BUGFIXES
+  - Regognize "CI_-" EnvVars (#6)
```
[the remark this answers is not in the document]
Comment (pat-s) [reply src/utils/__fixtures__/CHANGELOG_2.md:81]: jup

File: src/utils/__fixtures__/CHANGELOG_2.md [hunk 2]
```diff
@@ -2,84 +2,84 @@
 
 ## [v2.0.3](https://github.com/woodpecker-ci/plugin-git/releases/tag/v2.0.3) - 2022-12-29
 
-* BUGFIXES
-  * Fix write .netrc location with home var (#57)
+- BUGFIXES
+  - Fix write .netrc location with home var (#57)
```
[the remark this answers is not in the document]
Comment (pat-s) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: Correct. I searched for it but also couldn't find it. Or phrased differently: I think it's done here

https://github.com/woodpecker-ci/plugin-ready-release-go/blob/816eef09d9a479f1b616f9a98d4c980b1df74a78/src/utils/change.ts#L73-L76

but this uses already `-` and no `*`. So I don't know where the `*` come from. @anbraten might surely know!

[the remark this answers is not in the document]
Comment (anbraten) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: The [gitea changelog tool](https://gitea.com/gitea/changelog) uses `*` and I just copied them. So we might ignore those file or simply change them, I leave that up to you.

[the remark this answers is not in the document]
Comment (pat-s) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: We can also change the markdownlint rule. 

Where is the code part that pull in the changelog tool? And is it configurable?

[the remark this answers is not in the document]
Comment (anbraten) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: > Where is the code part that pull in the changelog tool? And is it configurable?

I just copied it from the main woodpecker repo I think

[the remark this answers is not in the document]
Comment (pat-s) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: Sorry, I still don't know *what* you copied 😓 What I meant: where is the code which determines if the plugin uses `*` or `-` for unordered lists?

[the remark this answers is not in the document]
Comment (anbraten) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: The plugin is always using `-`: https://github.com/woodpecker-ci/plugin-ready-release-go/blob/main/src/utils/change.ts#L73

The fixtures are for testing and I used `*` (copied from woodpecker-ci/woodpecker repo) to test if old changelog entries will remain untouched.

[the remark this answers is not in the document]
Comment (pat-s) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: OK got it now 😅 . That means the fixtures change won't be causing trouble.

[the remark this answers is not in the document]
Comment (anbraten) [reply src/utils/__fixtures__/CHANGELOG_2.md:6]: > OK got it now 😅 . That means the fixtures change won't be causing trouble.

Yes should be fine. We check that we don't change old entries in the changelog in the unit tests anyways.

**Review by anbraten — APPROVED** [review]

**Pull request merged**